src/shims/fs.rs

```diff
@@ -548,6 +553,255 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.set_last_error(ebadf)?;
         Ok((-1).into())
     }
+
+
+    fn mkdir(
+        &mut self,
+        path_op: OpTy<'tcx, Tag>,
+        mode_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("mkdir")?;
+
+        #[cfg(target_os = "linux")]
```

Comment:
Most likely this is wrong. `target_os` is the OS of the compiled Miri, but you probably want to determine the OS of the interpreted program, and that information is available through `tcx`. See for example [here](https://github.com/rust-lang/miri/blob/ad7b1bd8367b1dc141ae23033f195ef03e3ccde1/src/shims/fs.rs#L340).